packages/expo-notifications/ios/EXNotifications/Notifications/Categories/CategoriesModule.swift

```diff
@@ -4,45 +4,54 @@ import ExpoModulesCore
 import UIKit
 import MachO
 
-public class CategoriesModule: Module {
+open class CategoriesModule: Module {
   public func definition() -> ModuleDefinition {
     Name("ExpoNotificationCategoriesModule")
 
-    AsyncFunction("getNotificationCategoriesAsync") { (promise: Promise) in
-      UNUserNotificationCenter.current().getNotificationCategories { categories in
-        let existingCategories = categories.map { category in
-          return CategoryRecord(category)
-        }
-        promise.resolve(existingCategories)
-      }
+    AsyncFunction("getNotificationCategoriesAsync") {
+      let categories = await UNUserNotificationCenter.current().notificationCategories()
+      return filterAndSerializeCategories(Array(categories))
```

Comment:
```suggestion
      return filterAndSerializeCategories(categories)
```

the conversion from Set to Array seems unnecessary

Reply:
@douglowder what I'm saying is, `filterAndSerializeCategories` could accept a Set, not an Array. Then this won't be needed

Reply:
I believe we would then have to do the conversion on the result?

Reply:
Done